lib/queues.js

```diff
@@ -611,5 +610,22 @@ module.exports = class Queues {
     // Lazy load configuration.
     return this._ironium.configPromise;
   }
+
+  // Returns the maximum amount of jobs a queue should try to reserve.
+  // Every queue gets a share of reservation capacity to better deal
+  // with load spikes across queues.
+  getReserveCapacity() {
+    return this.configPromise
+      .then(config => {
+        const totalCapacity       = config.concurrency - this._activeJobs;
+        const maxCapacityPerQueue = Math.floor(totalCapacity / this._queues.size);
+        const capacity            = Math.min(totalCapacity, maxCapacityPerQueue);
```

Comment:
Concurrency = 50
Active jobs = 21
Queues = 30
Max queue capacity =  0
Capacity = 1

So this algorithm will only run concurrency - _queues.size jobs at once?

Fundamentally, the problem here is confusing queues for workload. The workload are the jobs in all the queues, not the queues themselves.